pkg/apis/observability-ui/v1alpha1/types.go

```diff
@@ -0,0 +1,120 @@
+// +groupName=observabilityui.rhobs
+// +kubebuilder:rbac:groups=observabilityui.rhobs,resources=observabilityuiplugins,verbs=list;get;watch
+// +kubebuilder:rbac:groups=observabilityui.rhobs,resources=observabilityuiplugins/status;observabilityuiplugins/finalizers,verbs=get;update
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ObservabilityUIPlugin defines an observability console plugin
+// +k8s:openapi-gen=true
+// +kubebuilder:resource
+// +kubebuilder:subresource:status
+type ObservabilityUIPlugin struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ObservabilityUIPluginSpec   `json:"spec,omitempty"`
+	Status ObservabilityUIPluginStatus `json:"status,omitempty"`
+}
+
+// ObservabilityUIPluginList contains a list of ObservabilityUIPlugin
+// +kubebuilder:resource
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type ObservabilityUIPluginList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []ObservabilityUIPlugin `json:"items"`
+}
+
+type UIPluginType string
+
+const (
+	TypeDashboards UIPluginType = "dashboards"
+)
+
+// Observabilityuipluginpec is the specification for desired state of ObservabilityUIPlugin.
+type ObservabilityUIPluginSpec struct {
+	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=dashboards
+	Type UIPluginType `json:"type"`
```

Comment:
I was experimenting with `ObservabilityUIPlugin` creation in an OpenShift cluster and even though the `type` is displayed as required (in UI), I can create a plugin with an empty value. 

Perhaps the `// +kubebuilder:validation:Enum=dashboards` should be applied to `UIPluginType` type (example [here](https://github.com/openshift/api/blob/master/operator/v1/types.go#L94))

Reply:
I think this is still problematic (enum required validation). Maybe the `ObservabilityUIPluginSpec` should not have `omitempty` for validation to happen and it should probably be `// +kubebuilder:validation:Enum=Dashboards`

Maybe we could also declare `Dashboards` as the default value, but I don't know what the future plans are.

Reply:
@tremes wdyt about the current version here?